### DISCUSSION TITLE
I am using latest java-client ver 8.3 with Appium 2.0.0-beta-66 to test automate a Flutter application. Limitation with gestures API

### DISCUSSION THREAD

**Discussion by vnuta9** [appium/appium#18623] [Q&A]
When I try to use TouchActions class as instructed in this lesson, I got this suggestion "Deprecated

Touch actions are deprecated. Please use W3C Actions instead or the corresponding extension methods for the driver (if available)."I tried to find examples and JavaDoc for the W3cActions java classes, I find almost nothing. What is the best course of action for me ?

I hope you can help me here.

**Comment by jlipps** [comment]
Look at the Selenium client java docs for the Actions class.
